financial-connections/src/main/java/com/stripe/android/financialconnections/di/FinancialConnectionsSubcomponentBuilderProvider.kt

```diff
@@ -0,0 +1,35 @@
+package com.stripe.android.financialconnections.di
+
+import com.airbnb.mvrx.ViewModelContext
+import com.stripe.android.financialconnections.screens.ConsentSubcomponent
+import com.stripe.android.financialconnections.ui.FinancialConnectionsSheetNativeActivity
+import javax.inject.Inject
+import javax.inject.Provider
+
+/**
+ * Wrapper that exposes [FinancialConnectionsSheetNativeComponent]'s registered
+ * subcomponent builders.
+ *
+ * This will be injected and exposed in
+ * [com.stripe.android.financialconnections.presentation.FinancialConnectionsSheetNativeViewModel]
+ *
+ */
+internal class FinancialConnectionsSubcomponentBuilderProvider @Inject constructor(
+    val consentSubComponentBuilder: Provider<ConsentSubcomponent.Builder>
+    // other subcomponent builder providers will be added here
+)
+
+/**
+ * Mavericks uses [com.airbnb.mvrx.MavericksViewModelFactory] to create ViewModel instances.
+ *
+ * Since these factories are static objects, we can't inject the subcomponent builder as usual.
+ *
+ * Instead, these factories offer access to a [ViewModelContext] instance.
+ *
+ * This helps to retrieve [FinancialConnectionsSubcomponentBuilderProvider] when from the
+ * [ViewModelContext] instance by transparently accessing the viewModel property.
+ */
+internal val ViewModelContext.subComponentBuilderProvider: FinancialConnectionsSubcomponentBuilderProvider
```

Comment:
this extension method seems to have some quite specific assumptions and couldn't be reused as is, could we have a more specific name such as  `ViewModelContext.financialConnectionsSubcomponentBuilderProvider`?

Reply:
👍 , wasn't planning on reusing initially (just planning on a single-activity component to N subcomponents per composable representing a step) but just in case - https://github.com/stripe/stripe-android/pull/5122/commits/b393aa32091f1cdbe36b629cb8ed1272dbb2c532